### DISCUSSION TITLE
为什么代码在linux上响应很慢，win上很快

### DISCUSSION THREAD

**Discussion by DreamMaoMao** [acheong08/EdgeGPT#449] [General]
同样的python版本，同样的EdgeGPT.py（2023年5月28日）版本,同一台笔记本，连接同样的热点，ip都一致，且在未使用vpn代理的前提下：

运行代码python -m EdgeGPT --cookie-file=cookie.json --style=creative 

然后提问一个：hello

-- win pwsh几秒内就会有响应，
-- win wsl上需要二十多秒，
-- 切换到相同笔记本的ubuntu物理机上会直接超时，甚至把超时设置到60s都不行

这个问题已经测试了二十多遍，会稳定的重现，不是偶然

**Comment by acheong08** [comment]
Speed is determined by network. I only have Linux so I can't exactly test how fast it is on Windows. This could be an issue with your specific linux installations

**Comment by DreamMaoMao** [comment]
这似乎是个很玄学的问题

我尝试换掉新买的华为路由器之后，ubuntu上也可以正常使用了
总的来说：
就是win在新和旧路由器都可以使用

ubuntu只能在旧的路由器上使用，新路由器会在wss连接上超时

暂时无从查起，只能推断是ubuntu23.04和路由器之间的通信有问题，而win能兼容
